openedx/core/djangoapps/discussions/tests/test_views.py

```diff
@@ -219,127 +248,152 @@ def _assert_defaults(self, response):
         data['lti_configuration'].update(pii_fields)
         response = self._post(data)
-        assert response.status_code == 400
+        assert response.status_code == status.HTTP_400_BAD_REQUEST
 
     @ddt.data(
         {"pii_share_username": True},
         {"pii_share_email": True},
         {"pii_share_email": True, "pii_share_username": True},
     )
-    def test_post_pii_fields(self, pii_fields):
+    def test_post_pii_fields_with_pii_disabled(self, pii_fields):
         """
-        Only if PII sharing is enabled should a user be able to set pii fields.
+        Test that when PII sharing is disabled for the course, user is not able
+        update PII settings for a course.
         """
-        data = self._setup_lti()
+        data = self._configure_lti_discussion_provider()
         data['lti_configuration'].update(pii_fields)
         with self._pii_sharing_for_course(enabled=False):
             response = self._post(data)
-            assert response.status_code == 400
+            assert response.status_code == status.HTTP_400_BAD_REQUEST
+            lti_configuration = self.get()['lti_configuration']
+            no_pii_fields_updated = [
+                lti_configuration.get(pii_field) != pii_value for pii_field, pii_value in pii_fields.items()
+            ]
+            assert all(no_pii_fields_updated)
+
+    @ddt.data(
+        {"pii_share_username": True},
+        {"pii_share_email": True},
+        {"pii_share_email": True, "pii_share_username": True},
+    )
+    def test_post_pii_fields_with_pii_enabled(self, pii_fields):
+        """
+        Test that when PII sharing is enabled for the course, user is able
+        update PII settings for the course.
+        """
+        data = self._configure_lti_discussion_provider()
+        data['lti_configuration'].update(pii_fields)
         with self._pii_sharing_for_course(enabled=True):
             response = self._post(data)
-            assert response.status_code == 200
+            assert response.status_code == status.HTTP_200_OK
+
+            lti_configuration = self.get()['lti_configuration']
+            all_pii_fields_updated = [
+                lti_configuration[pii_field] == pii_value for pii_field, pii_value in pii_fields.items()
+            ]
+            assert all(all_pii_fields_updated)
 
     @ddt.data(
-        True, False
+        True,
+        False
     )
     def test_get_pii_fields(self, pii_sharing):
         """
-        Only if PII is enabled should pii fields be returned.
+        Tests that when PII sharing is enabled, API included PII info.
         """
-        self._setup_lti()
+        self._configure_lti_discussion_provider()
         with self._pii_sharing_for_course(enabled=pii_sharing):
-            response = self._get()
-            data = response.json()
+            data = self.get()
             # If pii_sharing is true, then the fields should be present, and absent otherwise
```

Comment:
is this comment still relevant?

Reply:
yes - this data is not available if pii sharing flag is disabled.